contracts/DixelClubV2NFT.sol

```diff
@@ -156,25 +164,30 @@ contract DixelClubV2NFT is ERC721Enumerable, Ownable, SVGGenerator {
 
         uint256 length = list.length; // gas saving
         for (uint256 i; i != length; ) {
-            _whitelist.push(list[i]);
+            address curr = list[i]; // gas saving
+            PosAndC storage pc = _whitelistData[curr];
+
+            if (!_whitelistData[address(0)].hitted[curr]) {
```

Comment:
Could you explain why you used `mapping(address => bool) hitted;` here?
If you wanted to prevent adding the same wallet address twice in the `_whitelist` array, we could probably just check
 ```solidity
if(!pc.init) {
    _whitelist.push(curr);
    (pc.Position, pc.init) = (uint96(_whitelist.length) - 1, true);
}
```

Reply:
That's right! It's to check the duplicate address. Keeping the array short is also used to reduce read overhead.